src/requirements/checkers/majors/be.ts

```diff
@@ -1,44 +1,40 @@
 import { Course } from '../../types';
-import { courseMatchesCode, courseMatchesCodeOptions } from '../checkers-common';
+import { courseMatchesCodeOptions, includesWithSingleRequirement, includesWithSubRequirements } from '../checkers-common';
 
-const bioEngineeringEngineeringDistributions = [
-  (course: Course): boolean => courseMatchesCodeOptions(course, ['ENGRD 2020']),
-  (course: Course): boolean => courseMatchesCodeOptions(course, ['ENGRD 2600', 'ENGRD 2510'])
-];
-
-const bioEngineeringIntroBio = (course: Course): boolean => courseMatchesCodeOptions(
-  course, ['BIOMG 1350', 'BIOG 1440', 'BIOG 1445', 'BIOEE/BIOSM 1610']
+const bioEngineeringEngineeringDistributions = includesWithSubRequirements(
+  ['ENGRD 2020'],
+  ['ENGRD 2600', 'ENGRD 2510']
 );
 
-const bioEngineeringRequiredMajorCourses = [
-  (course: Course): boolean => courseMatchesCode(course, 'BIOG/BIOSM 1500'),
-  (course: Course): boolean => courseMatchesCode(course, 'BEE 3500'),
-  (course: Course): boolean => courseMatchesCode(course, 'BEE 3310'),
-  (course: Course): boolean => courseMatchesCode(course, 'BEE 3400'),
-  (course: Course): boolean => courseMatchesCode(course, 'BEE 3600'),
-  (course: Course): boolean => courseMatchesCode(course, 'BEE 4500')
-];
+const bioEngineeringIntroBio = includesWithSingleRequirement('BIOMG 1350', 'BIOG 1440', 'BIOG 1445', 'BIOEE/BIOSM 1610');
+
+const bioEngineeringRequiredMajorCourses = includesWithSubRequirements(
+  ['BIOG/BIOSM 1500'],
+  ['BEE 3500'],
+  ['BEE 3310'],
+  ['BEE 3400'],
+  ['BEE 3600'],
+  ['BEE 4500']
+);
 
-const bioEngineeringBioChemistry = [
-  (course: Course): boolean => courseMatchesCodeOptions(course, ['BIOMG 3300', 'BIOMG 3330', 'BIOMG 3350', 'BIOMG 3310']),
-  (course: Course): boolean => courseMatchesCode(course, 'BIOMG 3320')
-];
+const bioEngineeringBioChemistry = includesWithSubRequirements(
+  ['BIOMG 3300', 'BIOMG 3330', 'BIOMG 3350', 'BIOMG 3310'],
+  ['BIOMG 3320']
+);
 
 const bioEngineeringEngineeringStatistics = (course: Course): boolean => (
   courseMatchesCodeOptions(course, ['BEE 2220', 'ENGRD 2210', 'CHEME 3130', 'MSE 3030'])
   || courseMatchesCodeOptions(course, ['CEE 3040', 'ENGRD 2700'])
 );
```

Comment:
This requirement computation is wrong. The fix is delayed to the next diff to ensure zero change in generated requirement json.

Reply:
Fixed in #105

Reply:
approved!